pkg/util/exclude_kubernetes.go

```diff
@@ -0,0 +1,106 @@
+package util
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/afero"
+	yaml "gopkg.in/yaml.v2"
+	"sigs.k8s.io/kustomize/pkg/types"
+)
+
+// calls ExcludeKubernetesResource for each excluded resource. Returns after the first error.
+func ExcludeKubernetesResources(fs afero.Afero, basePath string, excludedResources []string) error {
+	for _, excludedResource := range excludedResources {
+		err := ExcludeKubernetesResource(fs, basePath, excludedResource)
+		if err != nil {
+			return errors.Wrapf(err, "excluding %s from %s", excludedResource, basePath)
+		}
+	}
+	return nil
+}
+
+// exclude the provided kubernetes resource file from the kustomization.yaml at basePath, or from bases imported from that.
+func ExcludeKubernetesResource(fs afero.Afero, basePath string, excludedResource string) error {
+	kustomizeYaml, err := fs.ReadFile(filepath.Join(basePath, "kustomization.yaml"))
+	if err != nil {
+		return errors.Wrapf(err, "read kustomization yaml in %s", basePath)
+	}
+
+	kustomization := types.Kustomization{}
+	err = yaml.Unmarshal(kustomizeYaml, &kustomization)
+	if err != nil {
+		return errors.Wrapf(err, "unmarshal kustomization yaml from %s", basePath)
+	}
+
+	excludedResource = strings.TrimPrefix(excludedResource, string(filepath.Separator))
+
+	newResources := []string{}
+	for _, existingResource := range kustomization.Resources {
+		if existingResource == excludedResource {
+			fmt.Printf("excluding %s from %s", excludedResource, basePath)
+		} else {
+			newResources = append(newResources, existingResource)
+		}
+	}
+
+	if len(newResources) != len(kustomization.Resources) {
+		kustomization.Resources = newResources
+
+		// write updated kustomization to disk - resource has been removed
+		kustomizeYaml, err = yaml.Marshal(kustomization)
+		if err != nil {
+			return errors.Wrapf(err, "marshal kustomization yaml from %s", basePath)
+		}
+
+		err = fs.WriteFile(filepath.Join(basePath, "kustomization.yaml"), kustomizeYaml, 0644)
+		if err != nil {
+			return errors.Wrapf(err, "write kustomization yaml to %s", basePath)
+		}
+		return nil
+	} else {
+		// check if the resource is already removed from this dir
+		alreadyRemoved := false
+		err = fs.Walk(basePath, func(path string, info os.FileInfo, err error) error {
```

Comment:
i think you should handle the error here. otherwise call to `info.IsDir()` will panic

```go
err = fs.Walk(basePath, func(path string, info os.FileInfo, err error) error {
    if err != nil {
        return err
    }
    ...
```

Reply:
That I can do
Hadn't realized that it passed in an error 🤦‍♂